main.cpp

```diff
@@ -248,6 +250,34 @@ EXPORT int HandleChat(wchar_t* msg) {
         CommandsModMessage(L"Teleporting.\n");
         return 1;
 
+    } else if ( swscanf(msg, L"/gui chat %d %d", &targetx, &targety) == 2) {
+        game->gui.chat_widget->width = targetx;
+        game->gui.chat_widget->height = targety;
+        CommandsModMessage(L"GUI Chat Widget size changed.\n");
+        return 1;
+
+    } else if ( !wcsncmp(msg, L"/tp ", 4) ) {
+        wchar_t* wideName = &msg[4];
+        char* cName = new char[wcslen(wideName)+1];
+        int len = wcstombs(cName, wideName, 0x100);
+        if (len < 1) {
+            CommandsModMessage(L"Invalid name.\n");
+            return 1;
+        }
+        for(cube::Creature* creature: game->world->creatures) {
+            if (stricmp(creature->name, cName) == 0) {

```

Comment:
Normally you'd use ! here, but if you want to keep it as == 0, I'll let it pass.